logger.go

```diff
@@ -181,32 +181,44 @@ func (logger *Logger) Println(args ...interface{}) {
 }
 
 func (logger *Logger) Warnln(args ...interface{}) {
-	if logger.Level >= WarnLevel {
+	if logger.level() >= WarnLevel {
 		NewEntry(logger).Warnln(args...)
 	}
 }
 
 func (logger *Logger) Warningln(args ...interface{}) {
-	if logger.Level >= WarnLevel {
+	if logger.level() >= WarnLevel {
 		NewEntry(logger).Warnln(args...)
 	}
 }
 
 func (logger *Logger) Errorln(args ...interface{}) {
-	if logger.Level >= ErrorLevel {
+	if logger.level() >= ErrorLevel {
 		NewEntry(logger).Errorln(args...)
 	}
 }
 
 func (logger *Logger) Fatalln(args ...interface{}) {
-	if logger.Level >= FatalLevel {
+	if logger.level() >= FatalLevel {
 		NewEntry(logger).Fatalln(args...)
 	}
 	os.Exit(1)
 }
 
 func (logger *Logger) Panicln(args ...interface{}) {
-	if logger.Level >= PanicLevel {
+	if logger.level() >= PanicLevel {
 		NewEntry(logger).Panicln(args...)
 	}
 }
+
+func (logger *Logger) level() Level {
+	logger.mu.RLock()
+	defer logger.mu.RUnlock()
+	return logger.Level
+}
+
+func (logger *Logger) formatter() Formatter {
+	logger.mu.RLock()
+	defer logger.mu.RUnlock()
+	return logger.Formatter
```

Comment:
You may avoid some overhead by not using `defer` as it is not strictly necessary here - reading a field can not panic and `logger` is not `nil` because it was accessed one line above. Same for the `level()` method.